internal/data/questions/questions.go

```diff
@@ -30,6 +35,102 @@ func GetQuestion(quizID uuid.UUID) (Question, error) {
 	return SampleQuestions[0], nil
 }
 
+// Returns all questions for a given quiz.
+// Includes the article for each question.
+// Includes the alternatives for each question.
+func GetQuestionsByQuizID(db *sql.DB, id uuid.UUID) (*[]Question, error) {
+	rows, err := db.Query(
+		`SELECT
+				q.id, q.question, q.arrangement, q.article_id, q.quiz_id, q.points,
+				array_agg(a.id)
+			FROM
+				questions q
+			LEFT JOIN
+				answer_alternatives a ON q.id = a.question_id
+			WHERE
+				quiz_id = $1
+			GROUP BY
+				q.id`,
+		id)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close() // Close the rows when the function returns
+
+	return scanQuestionsFromFullRows(db, rows)
+}
+
+// Converts a row from the database to a list of questions
+func scanQuestionsFromFullRows(db *sql.DB, rows *sql.Rows) (*[]Question, error) {
+	questions := []Question{}
+
+	for rows.Next() {
+		var q Question
+		var articleID uuid.UUID
+		var alternativeIDs []uuid.UUID
+		err := rows.Scan(
+			&q.ID, &q.Text, &q.Arrangement, &articleID, &q.QuizID, &q.Points,
+			pq.Array(&alternativeIDs),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add the article to the question
+		article, _ := articles.GetArticleByID(db, articleID)
```

Comment:
not sure this call should be here?
function name suggest it only scans a row into the type, but it also does a call to the db and gets the alternatives.

in my opinion this should be refactored and made clearer